### ISSUE TITLE
Clarification on `Location` in ABNF

### ISSUE THREAD

**Issue by SiriusStarr** [dhall-lang/dhall-lang#1264]
Currently the `Location` rule in the ABNF is listed under "Reserved identifiers":

**Comment by Gabriel439** [comment]
Yeah, it does not need to be treated as a reserved identifier since it's only used as part of an `as Location` modifier, so it can't be mistaken for an identifier
